src/scrapers/leumi.js

```diff
@@ -0,0 +1,176 @@
+import moment from 'moment';
+import { BaseScraperWithBrowser, LOGIN_RESULT } from './base-scraper-with-browser';
+import { dropdownSelect, fillInput, clickButton, waitUntilElementFound } from '../helpers/elements-interactions';
+import { waitForNavigation } from '../helpers/navigation';
+import { SHEKEL_CURRENCY, NORMAL_TXN_TYPE } from '../constants';
+
+const BASE_URL = 'https://hb2.bankleumi.co.il/';
+const DATE_FORMAT = 'DD/MM/YY';
+
+function getTransactionsUrl() {
+  return `${BASE_URL}/ebanking/Accounts/ExtendedActivity.aspx?WidgetPar=1#/`;
+}
+
+function getPossibleLoginResults() {
+  const urls = {};
+  urls[LOGIN_RESULT.SUCCESS] = /ebanking\/SO\/SPA.aspx/;
+  urls[LOGIN_RESULT.INVALID_PASSWORD] = /InternalSite\/CustomUpdate\/leumi\/LoginPage.ASP/;
+  // urls[LOGIN_RESULT.CHANGE_PASSWORD] = ``; // TODO should wait until my password expires
+  return urls;
+}
+
+function createLoginFields(credentials) {
+  return [
+    { selector: '#wtr_uid', value: credentials.username },
+    { selector: '#wtr_password', value: credentials.password },
+  ];
+}
+
+function getAmountData(amountStr) {
+  const amountStrCopy = amountStr.replace(',', '');
+  const amount = parseFloat(amountStrCopy);
+  const currency = SHEKEL_CURRENCY;
+
+  return {
+    amount,
+    currency,
+  };
+}
+
+function convertTransactions(txns) {
+  return txns.map((txn) => {
+    const txnDate = moment(txn.date, DATE_FORMAT).toISOString();
+
+    const credit = getAmountData(txn.credit).amount;
+    const debit = getAmountData(txn.debit).amount;
+    const amount = (Number.isNaN(credit) ? 0 : credit) - (Number.isNaN(debit) ? 0 : debit);
+    return {
+      type: NORMAL_TXN_TYPE,
+      identifier: txn.reference ? parseInt(txn.reference, 10) : null,
+      date: txnDate,
+      processedDate: txnDate,
+      originalAmount: amount,
+      originalCurrency: SHEKEL_CURRENCY,
+      chargedAmount: amount,
+      description: txn.description,
+      /* memo: txn.memo, TODO add this line to export transaction memo */
+    };
+  });
+}
+
+async function fetchTransactionsForAccount(page, startDate) {
+  // TODO provide actual start date
```

Comment:
@esakal what's the significance of this TODO? it seems like you ARE providing a start date.